### PR TITLE
get properties of class passed to geteAllowedAttributes rather than f…

### PR DESCRIPTION
**Pull request by jmwri**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2931 
| License       | MIT
| Doc PR        | 

I've documented the issue in #2931.

### CODE REVIEW

**Comment by jmwri** [comment]
@dunglas CI tests are failing due to memory issues. I assume this is a general issue and nothing to do with this PR?

File: src/Serializer/AbstractItemNormalizer.php [hunk 1]
```diff
@@ -322,11 +322,12 @@ protected function extractAttributes($object, $format = null, array $context = [
     protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
     {
         $options = $this->getFactoryOptions($context);
-        $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
+        $class = \is_object($classOrObject) ? \get_class($classOrObject) : $classOrObject;
```
Comment (soyuka) [inline src/Serializer/AbstractItemNormalizer.php]: it should use `$this->getObjectClass`

**Comment by jmwri** [comment]
Hi @soyuka,

I've updated the PR as per your comment :)

**Comment by soyuka** [comment]
can you try to rebase / make tests pass?

**Comment by teohhanhui** [comment]
Sorry, this does not make sense.

If you're asking for a `ResourceInterface`, and the `@type` in the response is `ResourceInterface`, then you're only supposed to get what's exposed by the `ResourceInterface` API resource. Otherwise we're leaking information and that's dangerous.

**Comment by jmwri** [comment]
I've replied in the related issue. Is this an issue of implementation? Would it make more sense to check if the `$class == $context['resource_class'] || is_subclass_of($class, $context['resource_class']`?

**Comment by teohhanhui** [comment]
@jmwri It's not a problem with your implementation. The current behaviour seems correct to me, and this PR actually will reintroduce an information leak. So the change is undesirable.

**Comment by parsingphase** [comment]
@teohhanhui Under what cases would you create a subclass of a displayable entity and not expect that subclass to be displayed? I appreciate the concern if this was a more general leak but I'd have to assume that users of the API would expect inheritors of entities to work this way.

**Comment by teohhanhui** [comment]
You can have any number of subclasses, and they shouldn't leak information unless you explicitly say that those subclasses should be exposed, by marking them as API resource classes. See discussion in https://github.com/api-platform/core/issues/2931#issuecomment-513138882

**Pull request closed without merging**